automated install/basic-install.sh

```diff
@@ -1377,8 +1377,29 @@ installConfigs() {
         fi
     fi
 
-    # Install pihole-FTL.service
-    install -T -m 0755 "${PI_HOLE_LOCAL_REPO}/advanced/Templates/pihole-FTL.service" "/etc/init.d/pihole-FTL"
+    # Install pihole-FTL systemd or init.d service, based on whether systemd is the init system or not
+    # Follow debhelper logic, which checks for /run/systemd/system to derive whether systemd is the init system
+    if [[ -d '/run/systemd/system' ]]; then
+        install -T -m 0644 "${PI_HOLE_LOCAL_REPO}/advanced/Templates/pihole-FTL.systemd" '/etc/systemd/system/pihole-FTL.service'
+
+        # Set net admin permissions so that FTL can serve DNS, DHCP and IMAP (for DHCPv6). If this does not work, run FTL as root user.
+        if ! setcap CAP_NET_BIND_SERVICE,CAP_NET_RAW,CAP_NET_ADMIN,CAP_SYS_NICE,CAP_IPC_LOCK,CAP_CHOWN+eip '/usr/bin/pihole-FTL'; then
+            sed -i '/^User=/d' '/etc/systemd/system/pihole-FTL.service'
+        fi
```

Comment:
I just realized: With `AmbientCapabilities=` in the systemd unit, setting capabilities on the binary is redundant, isn't it? The init.d script still does this internally, like it did before.

Reply:
What happens if we remove this part and `systemd` fails to set `AmbientCapabilities=`? Do we need a check  for this so FTL can run as `root`?

Reply:
Well, this is not a command one can test but a native systemd feature. If systemd fails to apply directives, the whole service fails (usually). Since a lot of systemd services shipped with systemd itself and other packages use this feature, the system would be broken if this didn't work, so IMO it's not reasonable to check for such basic systemd features. But I also don't know about a case where `setcap` fails (aside of missing permissions, R/O, immutable flag and such things, of course, which is then a mis-configuration on a different level).

@DL6ER since you did #2378
Do you remember the cases where this happened, and are such systems/kernels even able to run systemd as init system, or is it narrowed down custom build for embedded systems which run other init systems or only single applications at boot anyway?

Reply:
The issue really isn't with old kernels (the last 2.4 was released 2011 - such devices really shouldn't be connected to the Internet these days) but rather the other aspect: file systems. Capabilities are dependent on the file system supporting extended attributes (like Ext2, Ext3, Ext4, Btrfs, JFS, XFS, and Reiserfs) but many popular ones used by proprietary network file systems (FAT, NTFS) do not.

I think we discovered it back then because `docker` suffers (or suffer*ed* ?) from the issue that, when a volume is stored on these filesystems, capabilities weren't available either. I also remember some users reporting errors running Pi-hole on cheepy VPS where setting capabilities was prohibited for the user machines - either as a "security" measure or, more likely, because the administrators don't understand what capabilities are good for and that they can never grant you more rights than your `root` user has anyway.

Having said all that, we should also remember that the PR you mentioned is more than 4 years old. Some things might have changed but surely not all. I'd consider `AmbientCapabilities` to be the superior choice in terms of compatibility, however, the currently used hack of starting as `root` and having FTL recognize this and drop itself to user `pihole` once it did what it needs `root`-powers for (binding to low-number ports for DNS; DHCP, etc.) works amazingly well, too.

TL;DR: `systemd` is using [`libcap`](https://linux.die.net/man/3/libcap) to grant capabilities to the just started *process*. You need to be `root` to do that (`sudo systemctl ...`). This is a different approach than ours - we are adding these capabilities to the binary and rely on the kernel to see this when executing the binary.

Reply:
Okay, FAT and NTFS cannot be used as rootfs anyway, but in (probably older) Docker or container images, e.g. with aufs instead of overlayfs, on filesystem basis this probably still makes sense. But in a container, you won't run systemd, and the systemd `AmbientCapabilities` does not use capabilities on filesystem level, so I think we can remove the check and fallback to root user in systemd unit.

But it still makes sense to (try) `setcup` the binary in addition to `AmbientCapabilities`, so that manual FTP starts, for testing, debugging or whatever, remains possible without doing it as root or granting capabilities first.